src/DocumentTemplate/tests/test_DT_Try.py

```diff
@@ -0,0 +1,221 @@
+        html = self.doc_class(
+            '<dtml-call expr="dummy_list.append(110)">'
+            '<dtml-try>'
+            'Variable "name": '
+            '<dtml-var expr="not_defined"> '
+            '<dtml-finally>'
+            'Finally variable: '
+            '<dtml-call expr="dummy_list.pop(0)"> '
+            '</dtml-try>')
+        dummy_list = [100]
+
+        self.assertRaises(NameError, html, dummy_list=dummy_list)
+        # 100 got removed in the finally block.
+        self.assertEqual(dummy_list, [110])
+
+    def test_DT_Try__Try__06(self):
+        """It allows different exception handlers."""
+        html = self.doc_class(
+            '<dtml-try>'
+            'Variable "name": '
+            '<dtml-var expr="not_defined">'
+            '<dtml-except KeyError>'
+            'Exception variable: '
+            '<dtml-var expr="str(100)">'
+            '<dtml-except NameError>'
+            'Exception variable: '
+            '<dtml-var expr="str(110)">'
+            '</dtml-try>')
+        res = html()
+        expected = ('Exception variable: 110')
+        self.assertEqual(res, expected)
+
+    def test_DT_Try__Try__07(self):
+        """It catches errors with handler for base exception."""
+        html = self.doc_class(
+            '<dtml-try>'
+            'Variable "name": '
+            '<dtml-raise KeyError></dtml-raise>'
+            '<dtml-except NameError>'
+            'Exception variable: '
+            '<dtml-var expr="str(100)">'
+            '<dtml-except LookupError>'
+            'Exception variable: '
+            '<dtml-var expr="str(110)">'
+            '</dtml-try>')
+        res = html()
+        expected = ('Exception variable: 110')
+        self.assertEqual(res, expected)
+
+    def test_DT_Try__Try__08(self):
+        """It raises if no exception handler matches."""
+        html = self.doc_class(
+            '<dtml-try>'
+            'Variable "name": '
+            '<dtml-var expr="not_defined">'
+            '<dtml-except KeyError>'
+            'Exception variable: '
+            '<dtml-var expr="str(100)">'
+            '<dtml-except AttributeError>'
+            'Exception variable: '
+            '<dtml-var expr="str(110)">'
+            '</dtml-try>')
+        self.assertRaises(NameError, html)
+
+    def test_DT_Try__Try__09(self):
+        """It returns a dtml-return immediately."""
+        html = self.doc_class(
+            '<dtml-try>'
+            'Variable "name": '
+            '<dtml-return ret>'
+            '<dtml-raise KeyError></dtml-raise>'
+            '<dtml-except LookupError>'
+            'Exception variable: '
+            '<dtml-var expr="str(110)">'
+            '</dtml-try>')
+        res = html(ret='Return variable: 101')
+        expected = ('Return variable: 101')
+        self.assertEqual(res, expected)
+
+    def test_DT_Try__Try__10(self):
+        """It does not break with strings as exceptions."""
```

Comment:
But it raises a well defined exception if doing so.

Reply:
That is also true.